solara/__main__.py

```diff
@@ -361,16 +361,19 @@ def run(
     #         # window.show()
     #         webview.start(debug=True)
 
-    def open_browser():
-        while not failed and (server is None or not server.started):
+    def check_server_started():
+        while not failed and (server is None or not server.started) or not solara.server.server.is_ready(url):
```

Comment:
So I think I've solved my second issue
> if I run solara run solara.website.pages, I get an internal server error, because of:

This is due to the change in import order. Because we now import `solara.server.server`, which in turn imports `from . import app`, `app.py` runs before [this line](https://github.com/widgetti/solara/blob/38f268d63b4f8a9742fcb70d1726e9a6a9cefe0a/solara/__main__.py#L387). Then `app.apps` is empty, and the server can't start.

For a solution, we could import `solara.server.server` here instead of at the top, although I'm not sure if this could introduce a race condition of some sort.

```suggestion
        import solara.server.server
        
        while not failed and (server is None or not server.started) or not solara.server.server.is_ready(url):
```